src/cli/commands/files/add.js

```diff
@@ -18,16 +52,53 @@ module.exports = Command.extend({
     }
   },
 
-  run: (recursive, path) => {
-    var node = new IPFS()
-    path = process.cwd() + '/' + path
-    node.files.add(path, {
-      recursive: recursive
-    }, (err, stats) => {
+  run: (recursive, inPath) => {
+    let rs
+
+    inPath = checkPath(inPath, recursive)
+
+    glob(path.join(inPath, '/**/*'), (err, res) => {
       if (err) {
-        return console.log(err)
+        throw err
       }
-      console.log('added', bs58.encode(stats.Hash).toString(), stats.Name)
+      utils.getIPFS((err, ipfs) => {
+        if (err) {
+          throw err
+        }
+        const i = ipfs.files.add()
+        var filePair
+        i.on('data', (file) => {
+          console.log('added', bs58.encode(file.multihash).toString(), file.path)
+        })
+        i.once('end', () => {
+          return
+        })
+        if (res.length !== 0) {
+          const index = inPath.lastIndexOf('/')
+          async.eachLimit(res, 10, (element, callback) => {
+            const addPath = element.substring(index + 1, element.length)
+            if (fs.statSync(element).isDirectory()) {
+              callback()
+            } else {
+              rs = fs.createReadStream(element)
+              filePair = {path: addPath, stream: rs}
+              i.write(filePair)
+              callback()
+            }
+          }, (err) => {
+            if (err) {
+              throw err
+            }
+            i.end()
+          })
+        } else {
+          rs = fs.createReadStream(inPath)
+          inPath = inPath.substring(inPath.lastIndexOf('/') + 1, inPath.length)
+          filePair = {path: inPath, stream: rs}
+          i.write(filePair)
+          i.end()
```

Comment:
Isn't inPath being converted to the filename? Instead of trimming the full path till the `cwd`?

Ran an experiment and I can only get the filename, the path value should be the total path relative to where the process was executed.

```
> var path = '/a/b/c/d'
undefined
> path.substring(path.lastIndexOf('/') + 1, path.length)
'd'
```